lib/event.ts

```diff
@@ -0,0 +1,57 @@
+import { Config } from './config';
+
+export type EventHeaders = {
+	appName: string;
+	resource: string;
+	origin: string;
+	action: string;
+	destination: string;
+	schemaVersion: number;
+};
+
+export class Event {
+	private headers: EventHeaders;
+	constructor(
+		private name: string,
```

Comment:
If you don't change the value of private variables you can use "private readonly"  instead of "private"